### ISSUE TITLE
[feature request] add a delay after switching tool to start mining

### ISSUE THREAD

**Issue by FlashlightET** [zelythia/AutoTools#20]
i am removing a mountain on a server. i like how i can use this mod to automatically switch between my shovel and pickaxe(s) when i get to dirt or gravel patches, however since i'm doing this on a server and mining with haste ii and efficiency v, the blocks break immediately. since they are breaking within a couple frames after switching tools, some desync happens and the first block(s) mined with the switched tool reappear which gets kind of annoying. 

my proposal: a setting in the properties to add a delay from after the switch occurs to start mining. 0.2 seconds might be plenty if my ping averages 40ms.

also i saw the durability feature get added and id like to say thanks

**Comment by kierkat10** [comment]
honestly, this is a really good idea for this problem. it’s one of the only drawbacks of when i use this mod, because my tools will often just switch a ton if i’m mining a lot. very cool!

**Comment by kierkat10** [comment]
ok, so after some testing, this issue only occurs when you have "Keep selected slot" on. just a side note if the dev ever comes around to seeing this.

**Comment by FlashlightET** [comment]
oh i completely forgot about that option. that would make sense. i would still have to test if that fixes it on the server i play on though (im busy rn but it probably would fix it)

**Comment by kierkat10** [comment]
yeah… though it sucks that i have to turn that off since  usually have a slot of my hotbar always clear, so it’ll switch to that slot constantly. maybe the dev will put out an update that fixes the issue, who knows.

**Comment by zelythia** [comment]
I believe the problem is that after switching Items in the Inventory, Minecraft doesn't allow you to start Mining instantly. Unfortunately I haven't yet found a viable solution for fixing this. 
This is propably related to issue #11

**Comment by zelythia** [comment]
I just added an experimental option that may fix desyncs with the server by adding a 1 Tick = 50ms delay for breaking a block after switching tools. But I couldn't test it very extensively because the problem is really hard for me to prduce consistently
